### PR TITLE
Polish batching, remove `invalidateBatch` path

### PR DESCRIPTION
**Pull request by amiecorso**
BW-508

### CODE REVIEW

File: src/SpendPermissionManager.sol [hunk 1]
```diff
@@ -52,14 +52,14 @@ contract SpendPermissionManager is EIP712 {
...
+        uint48 period;
         /// @dev Timestamp this spend permission is valid after (unix seconds).
         uint48 start;
         /// @dev Timestamp this spend permission is valid until (unix seconds).
         uint48 end;
-        /// @dev Time duration for resetting used allowance on a recurring basis (seconds).
-        uint48 period;
         /// @dev Array of (token, allowance) tuples applied to this batch.
-        TokenAllowance[] tokenAllowances;
+        PermissionDetails[] tokenAllowances;
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:62]: Thoughts on this rename?
```suggestion
        PermissionDetails[] permissions;
```

File: src/SpendPermissionManager.sol [hunk 2]
```diff
@@ -77,17 +77,17 @@ contract SpendPermissionManager is EIP712 {
     );
 
     bytes32 constant PERMISSION_BATCH_TYPEHASH = keccak256(
-        "SpendPermissionBatch(address account,address spender,uint48 start,uint48 end,uint48 period,TokenAllowance[] tokenAllowances)TokenAllowance(address token,uint160 allowance)"
+        "SpendPermissionBatch(address account,address spender,uint48 start,uint48 end,uint48 period,PermissionDetails[] tokenAllowances)PermissionDetails(address token,uint160 allowance,uint256 salt,bytes extraData)"
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:80]:
```suggestion
        "SpendPermissionBatch(address account,uint48 period,uint48 start,uint48 end,PermissionDetails[] permissions)PermissionDetails(address spender,address token,uint160 allowance,uint256 salt,bytes extraData)"
```

File: src/SpendPermissionManager.sol [hunk 3]
```diff
@@ -77,17 +77,17 @@ contract SpendPermissionManager is EIP712 {
...
     );
 
     bytes32 constant TOKEN_ALLOWANCE_TYPEHASH =
-        keccak256("TokenAllowance(address token,uint160 allowance,address spender,uint256 salt)");
+        keccak256("PermissionDetails(address token,uint160 allowance,address spender,uint256 salt,bytes extraData)");
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:84]:
```suggestion
        keccak256("PermissionDetails(address spender,address token,uint160 allowance,uint256 salt,bytes extraData)");
```

**Review by ilikesymmetry — APPROVED** [review]
thoughts on changing `salt` to `bytes32` instead of `uint256`?

File: src/SpendPermissionManager.sol [hunk 4]
```diff
@@ -77,18 +77,15 @@ contract SpendPermissionManager is EIP712 {
     );
 
     bytes32 constant PERMISSION_BATCH_TYPEHASH = keccak256(
-        "SpendPermissionBatch(address account,address spender,uint48 start,uint48 end,uint48 period,TokenAllowance[] tokenAllowances)TokenAllowance(address token,uint160 allowance)"
+        "SpendPermissionBatch(address account,address spender,uint48 period,uint48 start,uint48 end,PermissionDetails[] permissions)PermissionDetails(address spender,address token,uint160 allowance,uint256 salt,bytes extraData)"
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:80]: duplicate spender in top-level struct
```suggestion
        "SpendPermissionBatch(address account,uint48 period,uint48 start,uint48 end,PermissionDetails[] permissions)PermissionDetails(address spender,address token,uint160 allowance,uint256 salt,bytes extraData)"
```

**Review by ilikesymmetry — APPROVED** [review]

File: src/SpendPermissionManager.sol (hunk 4 above)
Comment (amiecorso) [reply src/SpendPermissionManager.sol:80]: lol i'm a mess tonight

Comment (amiecorso) [reply src/SpendPermissionManager.sol:80]: I even carefully examined this

**Pull request merged**